Add publish and release as separate actions (#106)
bakdata/ci-templates: actions/tag-and-release/README.md

# tag-and-release

> This composite action allows you to tag a release a repository.

Reviewer: ```suggestion
This composite action allows you to create a Git tag and GitHub release.
```

## Dependencies

This action uses another composite action listed below:

- [action-gh-release](https://github.com/softprops/action-gh-release)

## Input Parameters

| Name          | Required | Default Value | Description                                                 |
| ------------- | :------: | :-----------: | ----------------------------------------------------------- |

> | tag           |    ✅    |       -       | The version of the tag to be publish and released e.g 1.0.0 |

Reviewer: description should match action
```suggestion
| tag           |    ✅    |       -       | The version of the tag to be publish and released, e.g. 1.0.0 |
```

> | release-title |    ✅    |       -       | Text for the github release title                           |

Reviewer: ```suggestion
| release-title |    ✅    |       -       | Title for the GitHub release                           |
```

> | release-body  |    ❌    |       -       | Description for the GitHub release                            |

Reviewer 2: What about `github-username`, `github-email` and `github-token`?

## Usage

```yaml
steps:
  - name: Check out repository
    uses: actions/checkout@v3

>   - name: Release to (Test)PyPI

Reviewer: wrong name

    uses: bakdata/ci-templates/actions/tag-and-release@main
    with:

>       tag: 1.0.0

Reviewer: ```suggestion
      tag: "1.0.0"
```

semver should always be parsed as string, otherwise it can break for versions such as 1.10 because it gets interpreted as float therefore it will be converted to 1.1

      release-title: "This should be the title for the github release"

>       release-body: "This should be on the text part of the github release"

Reviewer: capitalisation in GitHub
